restful-tango/tangoREST.py

```diff
@@ -301,12 +302,11 @@ def upload(self, key, courselab, file, body):
             labPath = self.getDirPath(key, courselab)
             try:
                 if os.path.exists(labPath):
+                    fileMD5 = hashlib.md5(body).hexdigest()
+                    filesInDir = self.computeMD5(labPath)
+                    if file in filesInDir and filesInDir[file] == fileMD5:
+                        return self.status.file_exists
```

Comment:
No, so previously we were not uploading the file if the md5 hash of the contents matched the md5 hash of one of the files already in the directory. This means that a file with a different name but the same contents would not be uploaded. 

Now, we only reject the upload if **both the filename and md5 hash** are the same. We actually check both the filename and the hash on the frontend, and we originally thought that was enough but it wasn't so we want to do this check in the backend as well.